morpho-checkpoint-node/src/Morpho/Tracing/Pretty.hs

```diff
@@ -1,81 +1,159 @@
+{-# LANGUAGE DataKinds #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE NamedFieldPuns #-}
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE QuantifiedConstraints #-}
+{-# LANGUAGE RankNTypes #-}
 {-# LANGUAGE RecordWildCards #-}
 {-# LANGUAGE UndecidableInstances #-}
 
 module Morpho.Tracing.Pretty where
 
 import Cardano.Crypto.DSIGN.Class (Signable)
-import Cardano.Crypto.Hash.Class (Hash, HashAlgorithm, hashToTextAsHex)
+import Cardano.Crypto.Hash.Class (Hash, hashToTextAsHex)
 import Cardano.Prelude hiding (list)
+import Cardano.Slotting.Slot (WithOrigin (At, Origin))
```

Comment:
The `cardano-slotting` library has been removed as a dependency, but we can get the same imports from

```haskell
import Ouroboros.Consensus.Block.Abstract (WithOrigin(..))
```

instead